docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,275 @@
+---
+description: StreamingFast Substreams sink files
+---
+
+# `substreams-sink-files` introduction
+
+### Overview
+
+The `substreams-sink-files` tool provides the ability to pipe data extracted from a blockchain to various types of files-based persistence solutions. A few available options include [BigQuery](https://cloud.google.com/bigquery), [Clickhouse](https://clickhouse.com), custom scripts and other related data storage and processing tools accepting a file format. For example, you could extract all of the ERC20, ERC721, and ERC1155 transfers from the Ethereum blockchain and persist the data to a files-based store.
+
+Substreams modules are created and prepared for the sink tool. After the sink tool begins running, automated ingestion tasks can be setup to have BigQuery, or other files-based storage solution, ingest the data.
+
+As an example, you could use `substreams-sink-files` to sink data in `JSONL` format to a [Google Cloud Storage (GCS)](https://cloud.google.com/storage) bucket and configure a BigQuery Transfer job to run every 15 minutes. The scheduled job ingests the new files found in the GCS bucket where the data, extracted by the Substreams, was written.
+
+### Accompanying code example
+
+[The accompanying Substreams module](#) associated with this documentation is responsible for extracting a handful of data fields from the Block object injected into the Rust-based map module. The sink tool processes the extracted blockchain data line-by-line and outputs the data to the files-based persistence mechanism you've chosen.
+
+The accompanying code example extracts four data points from the Block object and packs them into the `substreams.sink.files.v1` protobuf's data model. The data is passed to the protobuf as a single line of plain text, a `CSV` entry, or a `JSONL` element. StreamingFast is working on support for binary formats such as [Avro](https://avro.apache.org/), [Parquet](https://parquet.apache.org/), and others.
```

Comment:
> StreamingFast is working on support for binary formats such as [Avro](https://avro.apache.org/), [Parquet](https://parquet.apache.org/), and others.

Let's be let's affirmative here as we are not actively working on this anymore. Maybe we should turn into a call to actions, something like, "binary format would be possible but it's not supported today, contributions welcome" something like that,

Reply:
The content was rewritten and will be available in the next update.